### ISSUE TITLE
wyd support to react 19

### ISSUE THREAD

**Issue by eduardocque** [welldone-software/why-did-you-render#298]
Hi, 

After move to react 19 ( https://react.dev/blog/2024/04/25/react-19-upgrade-guide  / https://react.dev/blog/2024/04/25/react-19-upgrade-guide#libraries-depending-on-react-internals-may-block-upgrades), seems to be that is failing due the use of a internal method

![image](https://github.com/welldone-software/why-did-you-render/assets/35545218/1be069b9-b690-4e19-a9e3-b7752f492c3e)

is there a workaround meanwhile this is fixed ?

Regards

**Comment by eduardocque** [comment]
@vzaidman hi mate, can be possible support react 19 ?

**Comment by vzaidman** [comment]
Yes I looked into it. It's not trivial to support but I'll be working on it in the upcoming weeks.
Thanks for noticing

**Comment by eduardocque** [comment]
no worries, hopefully will be soon, for the moment had to disable this library due that fatal error, but i will keep an eye about future updates

thx for the help

**Comment by jo-adithya** [comment]
Hello, any update on this? I think I stumble upon similar issue. I got this error message:
`can't access property "ReactCurrentOwner", wdyrStore.React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED is undefined`

**Comment by vzaidman** [comment]
Not supported yet. It's very hard to support, I've spend many hours but it's still hard to get right. You can follow the progress here: https://github.com/welldone-software/why-did-you-render/pull/301

**Comment by jo-adithya** [comment]
Ahh I see, thanks for replying and your great work!

**Comment by vzaidman** [comment]
Released! See https://github.com/welldone-software/why-did-you-render/releases/tag/v10.0.0
